lib/prerender.js

```diff
@@ -11,23 +11,29 @@ const path = require('path');
 const chalk = require('chalk');
 
 class Prerender extends Plugin {
-  constructor(builtAppTree, { urls, indexFile, emptyFile }, ui) {
+  constructor(builtAppTree, { urls, indexFile, emptyFile }, ui, plugins) {
     super([builtAppTree], { name: 'prember', needsCache: false });
     this.urls = urls || [];
     this.indexFile = indexFile || 'index.html';
     this.emptyFile = emptyFile || '_empty.html';
     this.ui = ui;
+    this.plugins = plugins;
   }
 
   async listUrls(app, host) {
+    let visit = async (url) => {
+      return this._visit(app, host, url);
+    };
+
     if (typeof this.urls === 'function') {
-      let visit = async (url) => {
-        return this._visit(app, host, url);
-      };
-      return await this.urls(this.inputPaths[0], visit);
-    } else {
-      return this.urls;
+      this.urls = await this.urls(this.inputPaths[0], visit);
     }
+
+    for (let plugin of this.plugins) {
+      this.urls = this.urls.concat(await plugin(this.inputPaths[0], visit));
```

Comment:
Is this too much public API?

Reply:
I think it's appropriate to pass the same arguments we pass to the app's own `urls` function, so this looks fine.